src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -31,81 +101,659 @@ const ActorInit En_Nb_InitVars = {
+    if (!(gSaveContext.eventInf[4] & 8)) {
+        SubS_UpdateFlags(&this->stateFlags, EN_NB_FLAG_1 | EN_NB_FLAG_2, EN_NB_FLAG_1 | EN_NB_FLAG_2 | EN_NB_FLAG_4);
+    } else {
+        SubS_UpdateFlags(&this->stateFlags, EN_NB_FLAG_4, EN_NB_FLAG_1 | EN_NB_FLAG_2 | EN_NB_FLAG_4);
+    }
+    EnNb_ChangeAnim(this, EN_NB_ANIM_0);
+
+    return true;
+}
+
+s32 EnNb_ProcessScheduleOutput(EnNb* this, PlayState* play, ScheduleOutput* scheduleOutput) {
+    s32 success;
+
+    this->actor.flags |= ACTOR_FLAG_1;
+    this->actor.targetMode = 0;
+    this->stateFlags = EN_NB_FLAG_NONE;
+    this->unk_274 = 40.0f;
+
+    switch (scheduleOutput->result) {
+        default:
+            success = false;
+            break;
+
+        case EN_NB_SCH_1:
+        case EN_NB_SCH_3:
+        case EN_NB_SCH_4:
+            success = func_80BC0C0C(this, play, scheduleOutput);
+            break;
+
+        case EN_NB_SCH_2:
+            success = func_80BC0B98(this, play, scheduleOutput);
+            break;
+    }
+    return success;
+}
+
+s32 func_80BC0D08(EnNb* this, PlayState* play) {
+    return 0;
+}
+
+void EnNb_HandleSchedule(EnNb* this, PlayState* play) {
+    if ((this->schResultValue == EN_NB_SCH_1) || (this->schResultValue == EN_NB_SCH_2) ||
+        (this->schResultValue == EN_NB_SCH_3) || (this->schResultValue == EN_NB_SCH_4)) {
+        func_80BC0D08(this, play);
+    }
+
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.world.rot.y, 3, 0x2AA8);
+}
+
+void EnNb_FollowSchedule(EnNb* this, PlayState* play) {
+    ScheduleOutput scheduleResult;
+
+    this->timePathTimeSpeed = REG(15) + ((void)0, gSaveContext.save.daySpeed);
+
+    if (gSaveContext.eventInf[4] & 8) {
+        scheduleResult.result = EN_NB_SCH_1;
+        EnNb_ProcessScheduleOutput(this, play, &scheduleResult);
+        this->actor.shape.shadowDraw = ActorShadow_DrawCircle;
+        this->actor.flags |= ACTOR_FLAG_1;
+    } else if ((!Schedule_RunScript(play, sScheduleScript, &scheduleResult)) ||
+               ((this->schResultValue != scheduleResult.result) &&
+                !EnNb_ProcessScheduleOutput(this, play, &scheduleResult))) {
+        this->actor.shape.shadowDraw = NULL;
+        this->actor.flags &= ~ACTOR_FLAG_1;
+        scheduleResult.result = EN_NB_SCH_NONE;
+    } else {
+        this->actor.shape.shadowDraw = ActorShadow_DrawCircle;
+        this->actor.flags |= ACTOR_FLAG_1;
+    }
+
+    this->schResultValue = scheduleResult.result;
+    this->unk_1E8 = func_80BBFF90(this, play);
+    EnNb_HandleSchedule(this, play);
+}
+
+void func_80BC0EAC(EnNb* this, PlayState* play) {
+    if (func_8010BF58(&this->actor, play, this->unk_1E0, this->unk_28C, &this->unk_1E4)) {
+        if (gSaveContext.eventInf[4] & 8) {
+            gSaveContext.eventInf[4] &= (u8)~4;
+            gSaveContext.eventInf[4] &= (u8)~8;
+        }
+
+        SubS_UpdateFlags(&this->stateFlags, EN_NB_FLAG_1 | EN_NB_FLAG_2, EN_NB_FLAG_1 | EN_NB_FLAG_2 | EN_NB_FLAG_4);
```

Comment:
It is fine as is now, but just noting that these flags will probably end up being an enum from SubS